packages/server/api/src/app/workers/engine-controller.ts

```diff
@@ -135,15 +147,28 @@ export const flowEngineWorker: FastifyPluginAsyncTypebox = async (app) => {
       tags: runDetails.tags ?? [],
     });
 
-    if (populatedRun.status === FlowRunStatus.RUNNING) {
-      return;
+    const updateLogs =
+      !isNil(executionStateContentLength) && executionStateContentLength > 0;
+    if (!updateLogs) {
+      if (progressUpdateType === ProgressUpdateType.TEST_FLOW) {
+        app.io
+          .to(request.principal.projectId)
+          .emit(WebsocketClientEvent.TEST_FLOW_RUN_PROGRESS, runId);
+      } else {
+        app.io
+          .to(request.principal.projectId)
+          .emit(WebsocketClientEvent.FLOW_RUN_PROGRESS, runId);
```

Comment:
I wonder if we need both `TEST_FLOW_RUN_PROGRESS` and `FLOW_RUN_PROGRESS`, I would keep only `FLOW_RUN_PROGRESS` unless we want to have some telemetry based on them?

Reply:
I received info from Marco, that scheduled runs (production runs don't need live update). This will make it better from a technical perspective as we don't send too many websocket requests in case there are many concurrent scheduled runs.